plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codemodernizer/CodeTransformTelemetryManager.kt

```diff
@@ -0,0 +1,191 @@
+// Copyright 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.codemodernizer
+
+import com.intellij.openapi.components.service
+import com.intellij.openapi.project.Project
+import org.apache.commons.codec.digest.DigestUtils
+import software.amazon.awssdk.services.codewhispererruntime.model.TransformationStatus
+import software.aws.toolkits.jetbrains.services.codemodernizer.model.CustomerSelection
+import software.aws.toolkits.jetbrains.services.codemodernizer.model.JobId
+import software.aws.toolkits.jetbrains.services.codemodernizer.model.ValidationResult
+import software.aws.toolkits.jetbrains.services.codemodernizer.state.CodeModernizerSessionState
+import software.aws.toolkits.jetbrains.services.codemodernizer.state.CodeTransformTelemetryState
+import software.aws.toolkits.telemetry.CodeTransformApiNames
+import software.aws.toolkits.telemetry.CodeTransformCancelSrcComponents
+import software.aws.toolkits.telemetry.CodeTransformJavaSourceVersionsAllowed
+import software.aws.toolkits.telemetry.CodeTransformJavaTargetVersionsAllowed
+import software.aws.toolkits.telemetry.CodeTransformMavenBuildCommand
+import software.aws.toolkits.telemetry.CodeTransformPatchViewerCancelSrcComponents
+import software.aws.toolkits.telemetry.CodeTransformPreValidationError
+import software.aws.toolkits.telemetry.CodeTransformStartSrcComponents
+import software.aws.toolkits.telemetry.CodeTransformVCSViewerSrcComponents
+import software.aws.toolkits.telemetry.CodetransformTelemetry
+import software.aws.toolkits.telemetry.Result
+import java.time.Instant
+import java.util.Base64
+
+/**
+ * CodeModernizerTelemetry contains helper functions for common operations that require telemetry.
+ */
+class CodeTransformTelemetryManager(private val project: Project) {
```

Comment:
though since this is relatively stateless, i prefer `object` and just passing in `project` for the metrics where you need the job status

but either works